packages/data/src/data-manager.ts

```diff
@@ -285,112 +286,108 @@ export class DataManager implements DataManagerInterface {
           // Return rule errors if present
           if (Object.values(RuleError).includes(locationMatched as RuleError))
             return locationMatched as RuleError;
-        } else {
-          // Empty experience locations list or unset Site Area means there's no restriction for the location
-          locationMatched = true;
         }
       }
-      // Validate locationProperties against site area rules
-      if (!locationProperties || locationMatched) {
-        let audiences = [],
-          segments = [],
-          matchedAudiences = [],
-          matchedSegments = [],
-          audiencesToCheck: Array<ConfigAudience> = [];
+      if (!locationMatched) {
+        this._loggerManager?.debug?.(
+          'DataManager.matchRulesByField()',
+          MESSAGES.LOCATION_NOT_MATCH,
+          this._mapper({
+            locationProperties: locationProperties,
+            [experience?.locations
+              ? 'experiences[].variations[].locations'
+              : 'experiences[].variations[].site_area']:
+              experience?.locations || experience?.site_area || ''
+          })
+        );
+        return null;
+      }
 
-        if (visitorProperties) {
-          if (
-            Array.isArray(experience?.audiences) &&
-            experience.audiences.length
-          ) {
-            // Get attached transient and/or permnent audiences
-            audiences = this.getItemsByIds(
-              experience.audiences,
-              'audiences'
-            ) as Array<ConfigAudience>;
+      let audiences = [],
+        segments = [],
+        matchedAudiences = [],
+        matchedSegments = [],
+        audiencesToCheck: Array<ConfigAudience> = [],
+        audiencesMatched = true, // No audiences to check means there's no restriction for the audience
```

Comment:
here same, if you do not have visitorProperties, this will remain true but you want it true only if no audiences...

Reply:
But here, what happens if we have matched `locationProperties` while not providing `visitorProperties` at all? we shouldn't we return the patched experience?

Reply:
I think yes, definitely we should return matched experience if have matched locationProperties while not providing visitorProperties at all.